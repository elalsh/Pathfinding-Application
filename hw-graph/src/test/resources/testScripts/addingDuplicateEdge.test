# The expected number of edges should stay the same since the edge added is duplicate

# Create a graph
CreateGraph graph1

# Add nodes A, B, C and D
AddNode graph1 A
AddNode graph1 B
AddNode graph1 C
AddNode graph1 D

# add an edge from A -> B and C -> D
AddEdge graph1 A B edgeAB
AddEdge graph1 C D edgeCD

# Print the nodes in the graph and the outgoing edges for src nodes
ListNodes graph1
ListChildren graph1 A
ListChildren graph1 C

# Add a duplicate edge into the graph
AddEdge graph1 A B edgeAB

# Print the nodes in the graph after adding the duplicate edge
ListNodes graph1
ListChildren graph1 A
ListChildren graph1 C

