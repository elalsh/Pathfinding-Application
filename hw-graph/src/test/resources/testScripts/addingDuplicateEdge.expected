# The expected number of edges should stay the same since the edge added is duplicate

# Create a graph
created graph graph1

# Add nodes A, B, C and D
added node A to graph1
added node B to graph1
added node C to graph1
added node D to graph1

# add an edge from A -> B and C -> D
added edge edgeAB from A to B in graph1
added edge edgeCD from C to D in graph1

# Print the nodes in the graph and the outgoing edges for src nodes
graph1 contains: A B C D
the children of A in graph1 are: B(edgeAB)
the children of C in graph1 are: D(edgeCD)

# Add a duplicate edge into the graph
added edge edgeAB from A to B in graph1

# Print the nodes in the graph after adding the duplicate edge
graph1 contains: A B C D
the children of A in graph1 are: B(edgeAB)
the children of C in graph1 are: D(edgeCD)
